lib/core/src/receive_swap.rs

```diff
@@ -270,6 +307,87 @@ impl ReceiveSwapStateHandler {
             (_, Failed) => Ok(()),
         }
     }
+
+    async fn verify_lockup_tx(
+        &self,
+        receive_swap: &ReceiveSwap,
+        swap_update_tx: &SwapUpdateTxDetails,
+        verify_confirmation: bool,
+    ) -> Result<()> {
+        // looking for lockup script history to verify lockup was broadcasted
+        let script_history = self.lockup_script_history(receive_swap).await?;
+        let lockup_tx_history = script_history
+            .iter()
+            .find(|h| h.txid.to_hex().eq(&swap_update_tx.id));
+
+        match lockup_tx_history {
+            Some(history) => {
+                info!("swapper lockup found, verifying transaction content...");
+
+                let lockup_tx = utils::deserialize_tx_hex(&swap_update_tx.hex)?;
+                if !lockup_tx
+                    .txid()
+                    .to_hex()
+                    .eq(&lockup_tx_history.unwrap().txid.to_hex())
```

Comment:
```suggestion
                    .eq(&history.txid.to_hex())
```

We already have `history`, so no need to `unwrap()`.

Reply:
fixed